service-commands/scripts/userClockValues.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Find the creator nodes associated with a user and their respective clock values.
+ *
+ * Script usage: node userClockValues.js <size> <outFile>
+ */
+const axios = require('axios')
+const CreatorNode = require('@audius/libs/src/services/creatorNode')
+const { Command } = require('commander')
+
+function commaSeparatedList(value, dummyPrevious) {
+  return value.split(',')
+}
+
+const program = new Command()
+program
+  .usage('')
+  .option('-h, --handles <handles>', 'Audius handle', commaSeparatedList, [])
+  .option('-i, --user-ids <userIds>', 'Audius user id', commaSeparatedList, [])
+
+// export DISCOVERY_PROVIDER_ENDPOINT="https://discoveryprovider.audius.co"
+const discoveryProviderEndpoint = process.env.DISCOVERY_PROVIDER_ENDPOINT
+
+async function getUserByHandle(handle, discoveryProviderEndpoint) {
+  try {
+    return (
+      await axios({
+        url: `/v1/full/users/handle/${handle}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and handle: ${handle} with ${err}`
+    )
+  }
+}
+
+async function getUserById(userId, discoveryProviderEndpoint) {
+  try {
+    const resp = (
+      await axios({
+        url: `/users?id=${userId}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+
+    if (!resp) {
+      throw new Error(`Failed to find user with userId ${userId}`)
+    }
+
+    return resp
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and user id: ${userId} with ${err}`
+    )
+  }
+}
+async function getClockValues({
+  wallet,
+  creator_node_endpoint: creatorNodeEndpoint
+}) {
+  const primaryCreatorNode = CreatorNode.getPrimary(creatorNodeEndpoint)
+  const secondaryCreatorNodes = CreatorNode.getSecondaries(creatorNodeEndpoint)
+
+  if (!creatorNodeEndpoint) {
+    return {
+      primaryNode: '',
+      primaryClockValue: '',
+      secondaryNodes: [],
+      secondaryClockValues: []
+    }
+  }
+
+  return {
+    primaryNode: primaryCreatorNode,
+    primaryClockValue: await CreatorNode.getClockValue(
+      primaryCreatorNode,
+      wallet
+    ),
+    secondaryNodes: secondaryCreatorNodes,
+    secondaryClockValues: await Promise.all(
+      secondaryCreatorNodes.map(secondaryNode =>
+        CreatorNode.getClockValue(secondaryNode, wallet)
+      )
+    )
+  }
+}
+
+async function run() {
+  try {
+    const { handles, userIds } = parseArgsAndEnv()
+    const userClockValues = await Promise.all([
+      ...handles.map(async handle => {
```

Comment:
did you know that `map()` actually returns an array 😉 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

you can update this like:
```
await Promise.all([...handles.map(...)]

await Promise.all(handles.map(...))
```

Reply:
My bad, I remember changing this above, but forgot to do it here, thanks!

Reply:
I'm merging two map results into a single array here, this kind of looked like a cleaner way to do it. Let me know what you think, thanks!